go/arrow/array/data.go

```diff
@@ -29,13 +29,14 @@ import (
 
 // Data represents the memory and metadata of an Arrow array.
 type Data struct {
-	refCount  int64
-	dtype     arrow.DataType
-	nulls     int
-	offset    int
-	length    int
-	buffers   []*memory.Buffer  // TODO(sgc): should this be an interface?
-	childData []arrow.ArrayData // TODO(sgc): managed by ListArray, StructArray and UnionArray types
+	refCount   int64
+	dtype      arrow.DataType
+	nulls      int
+	offset     int
+	length     int
+	buffers    []*memory.Buffer  // TODO(sgc): should this be an interface?
+	childData  []arrow.ArrayData // TODO(sgc): managed by ListArray, StructArray and UnionArray types
+	dictionary *Data             // only populated for dictionary arrays
```

Comment:
maybe expand the comment  to elaborate that for dictionary arrays buffers will contain nullability and indexes  that reference elements in this variable? (and childdata would be empty)

Reply:
done